pkgs/development/haskell-modules/configuration-nix.nix

```diff
@@ -502,4 +502,10 @@ self: super: builtins.intersectAttrs super {
 
   # Tests require a browser: https://github.com/ku-fpg/blank-canvas/issues/73
   blank-canvas = dontCheck super.blank-canvas;
+
+  # cabal2nix generates a dependency on base-compat, which is the wrong version
+  base-compat-batteries = overrideCabal super.base-compat-batteries (drv: {
+    libraryHaskellDepends =
+      with pkgs.haskellPackages; [ base base-compat_0_10_1 ];
```

Comment:
The proper way to fix that issue is to pass the appropriate version of the package as an argument to the build expression, i.e. by using `override` or `overrideScope` (if necessary). If you look around in that file, you'll find plenty of examples how to do that.